From 3f3fe35177ec4fb301f365413f30f74807ffd364 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Mon, 19 Oct 2026 21:32:57 -0700
Subject: [PATCH] fixup! avalanche: add treasury contender voting, 50/25/25
 reward split, and default slate

Assert miner, staker and treasury payout scripts through their actual typed coinbase commitments. Keep all exact subsidy/share amounts and contextual reward checks. Focused staking suite passes with the separately regenerated genesis-derived ID fixture.

Tier: V2
---
 src/avalanche/test/stakingrewards_tests.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/avalanche/test/stakingrewards_tests.cpp b/src/avalanche/test/stakingrewards_tests.cpp
index aed41c1deb..a81eb48660 100644
--- a/src/avalanche/test/stakingrewards_tests.cpp
+++ b/src/avalanche/test/stakingrewards_tests.cpp
@@ -2,6 +2,7 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#include <primitives/v2_outputs.h>
 #include <avalanche/avalanche.h>
 #include <avalanche/peermanager.h>
 #include <avalanche/processor.h>
@@ -193,11 +194,11 @@ BOOST_FIXTURE_TEST_CASE(staking_rewards_policy_and_coinbase_split,
         BOOST_CHECK_EQUAL(cb.vout[0].nValue,
                           blockSubsidy - expectedStakerShare -
                               expectedTreasuryShare);
-        BOOST_CHECK(cb.vout[0].scriptPubKey == minerScript);
+        BOOST_CHECK(CoinbaseOutputPaysScript(cb.vout[0], minerScript));
         BOOST_CHECK_EQUAL(cb.vout[1].nValue, expectedStakerShare);
-        BOOST_CHECK(cb.vout[1].scriptPubKey == stakerScript);
+        BOOST_CHECK(CoinbaseOutputPaysScript(cb.vout[1], stakerScript));
         BOOST_CHECK_EQUAL(cb.vout[2].nValue, expectedTreasuryShare);
-        BOOST_CHECK(cb.vout[2].scriptPubKey == treasuryScript);
+        BOOST_CHECK(CoinbaseOutputPaysScript(cb.vout[2], treasuryScript));
 
         CBlockIndex blockIdx;
         blockIdx.phashBlock = &dummyBlockHash;
-- 
2.50.1 (Apple Git-155)

